src/openrct2/paint/map_element/map_element.c

```diff
@@ -234,13 +234,34 @@ static void sub_68B3FB(paint_session * session, sint32 x, sint32 y)
     session->SpritePosition.x = x;
     session->SpritePosition.y = y;
     session->DidPassSurface = false;
+    sint32 previousHeight = 0;
     do {
         // Only paint map_elements below the clip height.
         if ((gCurrentViewportFlags & VIEWPORT_FLAG_PAINT_CLIP_TO_HEIGHT) && (map_element->base_height > gClipHeight)) break;
 
         sint32 direction = map_element_get_direction_with_offset(map_element, rotation);
         sint32 height = map_element->base_height * 8;
 
+        if (height != previousHeight)
+        {
+            previousHeight = height;
+            session->PathElementOnSameHeight = 0;
+            rct_map_element * ittElement = map_element;
+            while (!map_element_is_last_for_tile(ittElement++))
```

Comment:
ittElement?

Reply:
Sorry, that was a shorthand I did not mean to push. I gave the variable a proper name in the next push.